wasm/section.go

```diff
@@ -268,11 +269,14 @@ func (s *SectionTypes) ReadPayload(r io.Reader) error {
 	if err != nil {
 		return err
 	}
-	s.Entries = make([]FunctionSig, int(count))
-	for i := range s.Entries {
-		if err = s.Entries[i].UnmarshalWASM(r); err != nil {
+
+	s.Entries = make([]FunctionSig, 0, getInitialCap(count))
+	for i := uint32(0); i < count; i++ {
+		sig := FunctionSig{}
```

Comment:
we prefer `var v T` to spell out the zero value.
```suggestion
		var sig FunctionSig
```